api/etcdserverpb/rpc.proto

```diff
@@ -1158,6 +1158,16 @@ message DowngradeResponse {
   string version = 2;
 }
 
+// DowngradeVersionTestRequest is used for test only. The version in
+// this request will be read as the WAL record version.If the downgrade
+// target version is less than this version, then the downgrade(online)
+// or migration(offline) isn't safe, so shouldn't be allowed.
+message DowngradeVersionTestRequest {
```

Comment:
Is it like dry-run change?

Reply:
Thanks for the information.

If I understand correctly, solution 2 is better than solution 1 because it’s difficult to ensure there are no misuse calls.  If it's dry-run thing, like pre-validation call, solution 1 is better. I vote for this change.

Just thinking that `DowngradeVersionTestRequest` name is kind of confusing to me. It's not about downgrade, it's just to inject a WAL record to impact the mininum version result. What about `MigrationTestRequest`?

Reply:
Resolve this comment as we discussed on the call.